packages/node_modules/@node-red/nodes/core/storage/23-watch.js

```diff
@@ -23,9 +23,13 @@ module.exports = function(RED) {
     var getAllDirs = function (dir, filelist) {
         filelist = filelist || [];
         fs.readdirSync(dir).forEach(file => {
-            if (fs.statSync(path.join(dir, file)).isDirectory() ) {
-                filelist.push(path.join(dir, file));
-                getAllDirs(path.join(dir, file), filelist);
+            try {
+                if (fs.statSync(path.join(dir, file)).isDirectory() ) {
+                    filelist.push(path.join(dir, file));
+                    getAllDirs(path.join(dir, file), filelist);
+                }
+            } catch (error) {
+                //should we raise an error?
```

Comment:
depends - if the file did actually exist and now doesn't we should really just emit whatever the "file deleted" event is - if anything else then not so sure as at this point we are just adding things to the list to be monitored.

Reply:
What i did before was deleting a file while the node was inspecting it. Maybe this will also happen, if drive is broken (read-error) ...
maybe call the error : "Can't access file"?

Reply:
yes - so depends if we can tell what sort of error. Can we try the above fix and just log out the error to start with and see a few examples (assuming we can force them)

Reply:
we can try

if u like, i can arrage a teams meeting and you can test :-)

Reply:
the fix above from @hardillb  changed the behavior to not crash node-red.
![image](https://user-images.githubusercontent.com/75374180/119955230-26b4fc80-bfa0-11eb-97ff-7acbd1afd901.png)

Reply:
@dceejay ... dont know how to simulate this except deleting on my machine. A readlock would not prevent FS to acces the file ...

Reply:
right so if it's an ENOENT then it has been deleted - so should create a real gone away event (ideally) - as presumably it was there a fraction of a second before... Now we need same for inaccessible

Reply:
Inaccessible ... can happen by 1) you try to read the file and disk is broken and 2) you have a network disconnect ... bot both i cant simulate.

Reply:
Would it be a bad idea to handle the ENOENT and if any other happens do it generically?

Reply:
agree

Reply:
@dceejay looking at the code ... maybe even the fs.readdirSync(dir).forEach can fail by something like this. Maybe you should also add a catch for this.

Reply:
We have a scope problem at the moment, the `getAllDirs()` function is outside the node definition. If we move the try/catch inside the node say round line 75 then we'll stop iterating over the directories at the fist missing one, which will leave others unwatched.

Reply:
Yes, thats a point. So it should skip the error and continue  .... guess my error is causing some more pain ;-/